compiler/src/lnntoamm/Expr.ts

```diff
@@ -84,8 +101,37 @@ export default abstract class Expr {
         let [int, constant] = Const.fromConstantsAst(work, metadata);
         generated.push(...int);
         expr = constant;
+        skipDotIfNext();
+      } else if (work.has('fncall')) {
+        work = work.get('fncall');
+        if (expr === null) {
+          let assignableList = work.get('assignablelist');
+          if (!assignableList.has('assignables') || assignableList.get('cdr').has(0)) {
+            console.log(assignableList, assignableList.has('assignables'), assignableList.get('cdr').has(0));
+            throw new Error(`unexpected token: found ${work.t.trim()} but it's not applied to a function`);
+          }
+          const [intermediates, res] = Expr.fromAssignablesAst(assignableList.get('assignables'), metadata);
+          generated.push(...intermediates);
+          expr = res;
+          skipDotIfNext();
+        } else {
+          // it's probably an HOF
```

Comment:
Not a higher order function, you mean a method call, right?

Reply:
it's most likely not a method call - that would get triggered under `work.has('variable')`. If that's *not* triggered, then that might mean that the ast we're looking at is `doSomething(1)(2)`, which is calling the fn that was returned from an HOF

Reply:
Ah, got it